core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -193,6 +194,13 @@ public RemoteClusterService getRemoteClusterService() {
         return transportService.getRemoteClusterService();
     }
 
+    /**
+     * Return a map of nodeId to pending number of requests for the given action name prefix
+     */
+    public Map<String, Long> getPendingRequests(final String actionNamePrefix) {
```

Comment:
I wonder if we can keep a map inside `SearchTransportService` that is basically passed to every relevant request as an action listener. I think we can just keep things in the map until the counter goes to 0 and then we remove it?